### PR TITLE
added branching

### PR DESCRIPTION
**Pull request by Agwen12**

### CODE REVIEW

File: Fungi growth simulation/Assets/Direction.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static List<Direction> GetNeighbors(Direction direction)
         Vector3 directionVector3 = ToVector3(direction);
         float[] coords = { 1, 1, 1 };
 
-        if (Mathf.Abs(directionVector3.x) + Mathf.Abs(directionVector3.y) + Mathf.Abs(directionVector3.z) == 1) // 4 sides
+        if (DirectionMethods.IsSquare(directionVector3)) // 4 sides
```
Comment (Agwen12) [inline Fungi growth simulation/Assets/Direction.cs:68]: I've added method that check whether a direction point to a square face

File: Fungi growth simulation/Assets/Direction.cs [hunk 2]
```diff
@@ -113,4 +113,52 @@ public static Direction GetRandomDirection(Direction direction)
         List<Direction> neighbors = GetNeighbors(direction);
         return neighbors[Helper.Rnd.Next(neighbors.Count)];
     }
+
+
+    public static bool IsSquare(Vector3 vector) {
```
Comment (Agwen12) [inline Fungi growth simulation/Assets/Direction.cs:118]: method mentioned before

Comment (proman3419) [reply Fungi growth simulation/Assets/Direction.cs:118]: Can be private

File: Fungi growth simulation/Assets/Direction.cs [hunk 3]
```diff
@@ -113,4 +113,52 @@ public static Direction GetRandomDirection(Direction direction)
...
+        return vector.sqrMagnitude == 1;
+    }
+
+    public static Vector3[] GetBranchDirection(Direction direction) {
```
Comment (proman3419) [inline Fungi growth simulation/Assets/Direction.cs:122]: What if we reused the GetNeighbors method, then shuffled the returned list and took 2 elements from it?
[shuffle](https://www.delftstack.com/howto/csharp/shuffle-a-list-in-csharp/)
[take](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.take?view=net-6.0)

**Review by proman3419 — COMMENTED** [review]
Consider my suggestions

**Review by proman3419 — APPROVED** [review]

**Pull request closed without merging**